geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/QueryPlanner.scala

```diff
@@ -210,6 +213,33 @@ case class QueryPlanner(schema: String,
     }
   }
 
+  def adaptIteratorForMapAggregationQuery(accumuloIterator: CloseableIterator[Entry[Key, Value]],
+                                          query: Query,
+                                          returnSFT: SimpleFeatureType,
+                                          decoder: SimpleFeatureDecoder): CloseableIterator[SimpleFeature] = {
+    val aggregateKeyName = query.getHints.get(MAP_AGGREGATION_KEY).asInstanceOf[String]
+
+    val maps = accumuloIterator.map { kv =>
+      decoder
+        .decode(kv.getValue.get)
+        .getAttribute(aggregateKeyName)
+        .asInstanceOf[JMap[AnyRef, Int]]
+        .asScala
+    }
+
+    if(maps.nonEmpty) {
+      val reducedMap = sumNumericValueMutableMaps(maps.toIterable).toMap // to immutable map
```

Comment:
I'm trying to figure out the .toIterable and the implicit conversion happen with the calls to Iterator below.  

I think there may be a little bit of clean-up we can do here.